src/sentry/issues/producer.py

```diff
@@ -108,7 +108,13 @@ def _prepare_status_change_message(
         from sentry.issues.ingest import process_occurrence_data
 
         process_occurrence_data(status_change.to_dict())
-        group = get_group_from_fingerprint(status_change.to_dict())
+        fingerprint = status_change.fingerprint
```

Comment:
interesting callout, but I think it _is_ working - confirmed with Richard who's using this API.

Reply:
I have a feeling the line above can be removed with no impact because this is for the producer. So the raw fingerprint still gets passed to the occurrence consumer where it runs `process_occurrence_data` on the message correctly which happens at https://github.com/getsentry/sentry/blob/32cd372630013018538916c4d1776bb2bd816e29/src/sentry/issues/status_change_consumer.py#L165

Reply:
The line above is for a specific case in testing, where we don't actually use Kafka for the eventstream: https://github.com/getsentry/sentry/blob/32cd372630013018538916c4d1776bb2bd816e29/src/sentry/issues/producer.py#L106-L107

In that case, we want to still do the logic to change the status but without going through Kafka. That block will return None, which then returns out of the top-level `produce_occurrence_to_kafka` without actually producing a Kafka message.
https://github.com/getsentry/sentry/blob/32cd372630013018538916c4d1776bb2bd816e29/src/sentry/issues/producer.py#L55-L56

We have plans to refactor soon this so we don't maintain a second copy of essentially the same logic, and generate a mock Kafka message instead.